### PR TITLE
Few fixes

### PR DESCRIPTION
**Pull request by ruthenium96**

### CODE REVIEW

File: tests/CMakeLists.txt
Comment (joeydumont) [inline tests/CMakeLists.txt:1]: Could you make the `add_directory(tests)` in the main CMakeLists.txt conditional on BUILD_TESTING being true instead? That way, we wouldn't skip tests when we actually want to test them, and armadillo becomes a dev dependency.

**Review by joeydumont — CHANGES_REQUESTED** [review]
Thanks for your contribution, it is appreciated! 

Please note the changes I suggested in the review.

File: tests/CMakeLists.txt
Comment (ruthenium96) [reply tests/CMakeLists.txt:1]: I made the requested changes in the last commit. Thank you for this suggestion!

**Review by joeydumont — APPROVED** [review]
Sounds good, thanks!

**Pull request merged**